bpf/init.sh

```diff
@@ -304,12 +304,10 @@ function xdp_load()
 	NODE_MAC="{.addr=$(mac2array $NODE_MAC)}"
 
 	bpf_compile $IN $OUT obj "$OPTS -DNODE_MAC=${NODE_MAC}"
-
-	ip link set dev $DEV $MODE off
 	rm -f "$CILIUM_BPF_MNT/xdp/globals/$CIDR_MAP" 2> /dev/null || true
 	cilium-map-migrate -s $OUT
 	set +e
-	ip link set dev $DEV $MODE obj $OUT sec $SEC
+	ip -force link set dev $DEV $MODE obj $OUT sec $SEC
```

Comment:
Maybe it would make sense to write a human readable error to stderr if the cmd fails?

Reply:
We build iproute2 against libmnl, so in case XDP cannot be attached, the netlink message will contain an error message mentioning why/where it failed in the driver. Other than that for plain BPF load, it's the same as with the tc based one.

Reply:
These are marked via NL_SET_ERR_MSG() in the core (https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/net/core/dev.c#n8663) and in drivers.